frontend-maven-plugin/src/main/java/com/github/eirslett/maven/plugins/frontend/mojo/InstallNodeAndNpmMojo.java

```diff
@@ -101,6 +103,41 @@ public void execute(FrontendPluginFactory factory) throws Exception {
 
         String validNodeVersion = getDownloadableVersion(nodeVersion);
 
+        final String nodeDownloadRoot = getNodeDownloadRoot();
+        final String npmDownloadRoot = getNpmDownloadRoot();
+
+        if (isAtlassianProject(project) &&
+                isBlank(serverId) &&
+                (isBlank(nodeDownloadRoot) || isBlank(npmDownloadRoot))
+            ) { // If they're overridden the settings, they be the boss
+            getLog().info("Atlassian project detected, going to use the internal mirrors (requires VPN)");
+
+            serverId = "maven-atlassian-com";
+            try {
+                install(factory, validNodeVersion,
+                        isBlank(nodeDownloadRoot) ? ATLASSIAN_NODE_DOWNLOAD_ROOT : nodeDownloadRoot,
+                        isBlank(npmDownloadRoot) ? ATLASSIAN_NPM_DOWNLOAD_ROOT : npmDownloadRoot);
+                return;
+            } catch (InstallationException exception) {
+                // Ignore as many filesystem exceptions unrelated to the mirror easily
+                if (!(exception.getCause() instanceof DownloadException ||
+                        exception.getCause() instanceof ArchiveExtractionException)) {
+                    throw exception;
+                }
+                getLog().warn("Oh no couldn't use the internal mirrors! Falling back to upstream mirrors");
```

Comment:
this will risk that we silently don't know that something is misconfigured.

Maybe instead we can just override the roots and let it execute like it always did? (ie. just configure it automatically but don't change behavior nor the code even too much then)

Reply:
I don't think it's a real risk. This is an optimistic configuration. What I think is a real risk is that there's a public project we're working on which would fail because its CI will never be able to access the internal Maven registries. Similarly if PAC is ever having a bad day or if the version isn't populated yet, we want builds to be able to continue as much as possible.

We're not overriding the configuration if the user has explicitly set it, but we're optimistically trying when they haven't set it. This makes the onboarding to the fork easy.